### ISSUE TITLE
Implement NTK scorer 

### ISSUE THREAD

**Issue by mdbenito** [appliedAI-Initiative/pyDVL#274]
Introduced in _Zhaoxuan Wu, Yao Shu, and Bryan Kian Hsiang Low, “[DAVINZ: Data Valuation Using Deep Neural Networks at Initialization](https://proceedings.mlr.press/v162/wu22j.html),” in Proceedings of the 39th International Conference on Machine Learning (International Conference on Machine Learning, PMLR, 2022), 24150–76._

It's a scorer using the Neural Tangent Kernel with a correction to account for ditribution shift between training and validation set.
